### ISSUE TITLE
Review of the Pros/Cons of Neural Network training using Sequential Filtering vs. Gradient Descent

### ISSUE THREAD

**Issue by rameshnatarajanus** [YCCS-Summer-2023-DDNMA/project#101]
[Link to code](https://github.com/probml/dynamax/blob/fa4723f8e78aa52dc02859d45ce538464cee13ab/dynamax/utils/optimize.py#L4)

This is basically the most advanced issues from this collection of issues in using dynamax for neural network training.  Train the network using gradient descent and using EKF/UKF and illustrate the pros and cons of each approach.   

-- Which approach would you use if you had a large data set, versus a small dataset with the promise of more data being collected over time?
-- Are the two trained models identical, and are you able to recover the ground truth (viz. the model that was used to obtain the synthetic data for model training).

**Comment by JosephCuz** [comment]
look at multi-class versions of avi's model
